src/Badge.js

```diff
@@ -1,46 +1,56 @@
 import classNames from 'classnames';
 import React from 'react';
 
-import { prefix } from './utils/bootstrapUtils';
-import ValidComponentChildren from './utils/ValidComponentChildren';
-
-const Badge = React.createClass({
-  propTypes: {
-    pullRight: React.PropTypes.bool
-  },
-
-  getDefaultProps() {
-    return {
-      pullRight: false,
-      bsClass: 'badge'
-    };
-  },
+import { bsClass, getClassSet, omitBsProps } from './utils/bootstrapUtils';
 
-  hasContent() {
-    const { children } = this.props;
+const propTypes = {
+  pullRight: React.PropTypes.bool,
+};
 
-    return (
-      ValidComponentChildren.count(children) > 0 ||
-      React.Children.count(children) > 1 ||
-      typeof children === 'string' ||
-      typeof children === 'number'
-    );
-  },
+const defaultProps = {
+  pullRight: false,
+};
+
+class Badge extends React.Component {
+  hasContent(children) {
+    let result = false;
+
+    React.Children.forEach(children, child => {
+      if (result) {
+        return;
+      }
+
+      if (child != null && child !== false) {
```

Comment:
`false != null` is already true. Did you mean to write `!==`? Else I think you can use only the first conditions